integrated_channels/canvas/exporters/learner_data.py

```diff
@@ -103,10 +103,14 @@ def get_learner_assessment_data_records(
             if not subsection_percent_grade or not subsection_id:
                 continue
 
+            course_catalog_client = get_course_catalog_api_service_client(
+                site=enterprise_enrollment.enterprise_customer_user.enterprise_customer.site
+            )
+
             transmission_audit = CanvasLearnerAssessmentDataTransmissionAudit(
                 enterprise_course_enrollment_id=enterprise_enrollment.id,
                 canvas_user_email=enterprise_enrollment.enterprise_customer_user.user_email,
-                course_id=enterprise_enrollment.get_course_id(enterprise_enrollment.course_id),
+                course_id=course_catalog_client.get_course_id(enterprise_enrollment.course_id),
```

Comment:
aargh okay this makes some sense! so we get course if from course catalog client instead of the actuat enrollment? not clear why they differ but if this is the correct approach let's do it! just curious if you have thoughts on what the difference is

Reply:
The enrollment doesn't have any method of retrieving the course key haha, it was always the catalog with the course level reporting. I must have just misread it 😬. The catalog has a `get_course_run` method that contains the course key and which appears to be coming from the `course api utils`. Not sure if the enrollment would have access to this, so I just think the original way was just incorrect 🙃

Reply:
My only other thought is should be attempt to retrieve it before constructing the transmission audit and raise an error if it's unable to find anything? I think the Django model should raise a stink because the field would be null. So I guess it comes down to readability and debugging purposes, thoughts?